dali/operators/reader/loader/webdataset_loader.cc

```diff
@@ -255,12 +257,13 @@ void WebdatasetLoader::ReadSample(vector<Tensor<CPUBackend>>& sample) {
     current_wds_shard->Seek(component.offset);
 
     // Skipping cached samples
-    const std::string source_info =
+    const std::string image_key =
         make_string_delim(" ", "Archive:", paths_[current_sample.wds_shard_index],
                     GetSampleSource(current_sample), "; Component offset:", component.offset);
     DALIMeta meta;
-    meta.SetSourceInfo(source_info);
-    if (ShouldSkipImage(source_info)) {
+    meta.SetSourceInfo(make_string_delim(':', paths_[current_sample.wds_shard_index],
+                                         component.offset, component.filename));
+    if (ShouldSkipImage(image_key)) {
```

Comment:
This is a bug. The cache key MUST match the source info or the cache simply won't work - the decoded images are stored in the cache by a name obtained from GetSoruceInfo.

I guess the old formula for the image_key is no longer relevant. I can vaguely remember a moment when the index file contained less information and that's when GetSampleSource was invented. It looks obsolete now, but another pair of eyes would be useful here.

Reply:
Fixed